packages/backend/src/modules/tracked-txs/TrackedTxsModule.ts

```diff
@@ -0,0 +1,98 @@
+import { BigQuery } from '@google-cloud/bigquery'
+import { Logger } from '@l2beat/backend-tools'
+import { notUndefined } from '@l2beat/shared-pure'
+
+import { Config } from '../../config'
+import { BigQueryClient } from '../../peripherals/bigquery/BigQueryClient'
+import { Database } from '../../peripherals/database/Database'
+import { IndexerStateRepository } from '../../peripherals/database/repositories/IndexerStateRepository'
+import { Clock } from '../../tools/Clock'
+import {
+  ApplicationModule,
+  ApplicationModuleWithIndexer,
+} from '../ApplicationModule'
+import { HourlyIndexer } from './HourlyIndexer'
+import { createLivenessModule } from './modules/liveness/LivenessModule'
+import { TrackedTxsConfigsRepository } from './repositories/TrackedTxsConfigsRepository'
+import { TrackedTxsClient } from './TrackedTxsClient'
+import { TrackedTxsIndexer } from './TrackedTxsIndexer'
+
+export function createTrackedTxsModule(
+  config: Config,
+  logger: Logger,
+  database: Database,
+  clock: Clock,
+): ApplicationModuleWithIndexer<TrackedTxsIndexer> | undefined {
+  if (!config.trackedTxsConfig) {
+    logger.info('TrackedTxsModule disabled')
+    return
+  }
+
+  const indexerStateRepository = new IndexerStateRepository(database, logger)
+  const trackedTxsConfigsRepository = new TrackedTxsConfigsRepository(
+    database,
+    logger,
+  )
+
+  const hourlyIndexer = new HourlyIndexer(logger, clock)
+
+  const bigQuery = new BigQuery({
+    credentials: {
+      client_email: config.trackedTxsConfig.bigQuery.clientEmail,
+      private_key: config.trackedTxsConfig.bigQuery.privateKey,
+    },
+    projectId: config.trackedTxsConfig.bigQuery.projectId,
+  })
+  const bigQueryClient = new BigQueryClient(bigQuery)
+
+  const trackedTxsClient = new TrackedTxsClient(bigQueryClient)
+
+  const runtimeConfigurations = config.projects
+    .flatMap((project) => project.trackedTxsConfig?.entries)
+    .filter(notUndefined)
+
+  const livenessModule = createLivenessModule(config, logger, database, clock)
+  const subModules: (ApplicationModule | undefined)[] = [livenessModule]
+
+  const updaters = {
+    liveness: livenessModule?.updater,
+  }
+
+  const trackedTxsIndexer = new TrackedTxsIndexer(
+    logger,
+    hourlyIndexer,
+    updaters,
+    trackedTxsClient,
+    indexerStateRepository,
+    trackedTxsConfigsRepository,
+    runtimeConfigurations,
+    config.trackedTxsConfig.minTimestamp,
+  )
+
+  const start = async () => {
+    logger = logger.for('TrackedTxsModule')
+    logger.info('Starting...')
+
+    const areUpdatersEmpty = Object.values(updaters).every((u) => !u)
```

Comment:
IMO this code should not be there, in such case TrackedTxIndexer will have empty configurations and there is a code to handle that already. This line of code adds a false notion that TrackedTxsIndexer relies on updaters - but it  rather relies on configurations

Reply:
good catch